src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs

```diff
@@ -209,6 +209,18 @@ public class SourceRepositoryDependencyProvider : IRemoteDependencyProvider
             {
                 await EnsureResource();
 
+                // first check if the exact min version exist then simply return that
+                if (await _findPackagesByIdResource.DoesPackageExistAsync(
```

Comment:
`VersionRange.MinVersion` can be `null`, and `DoesPackageExistAsync(…)` will throw if `version` is `null`.  Although this code assumes `libraryRange.VersionRange.MinVersion` will never be non-null, I see no such guarantee.  Can you check that `libraryRange.VersionRange.MinVersion` is not `null`?